### PR TITLE
Blog Post: Upgrading from OpenDistro to OpenSearch

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Blog post on Upgrading from OpenDistro to OpenSearch.

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown [hunk 1]
```diff
@@ -0,0 +1,250 @@
+---
+layout: post
+title:  How To: Upgrade from Open Distro to OpenSearch
```
Comment (stockholmux) [inline _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:3]: The colon is causing a YAML parsing error 🤦‍♂️

Can you enclose the title in double quotes?

**Review by stockholmux — CHANGES_REQUESTED** [review]

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown (hunk 1 above)
Comment (saratvemulapalli) [reply _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:3]: Ah sure that make sense.

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+For more information on upgrading please see our [Upgrading FAQs](https://opensearch.org/faq/#c3).and if you have questions about backward compatibility you can check dB’s post on [Backwards Compatibility FAQ](https://opensearch.org/blog/technical-posts/2021/06/opensearch-backwards-compatibility-faq/).. If you’re looking for a great “getting started” guide I’d recommend Gedalyah’s [Install and Configure OpenSearch guide](https://logz.io/blog/opensearch-tutorial-installation-configuration/).
+
+Closing on a high note, hope this blog post helps and feel free to reach out if you have more questions. Thanks to OpenSearch team for making this happen.
+
+If you have any feedback/suggestions on new topics please reach out to us. we plan to have a follow up post to upgrade from Elasticsearch to OpenSearch for community who do not use Open Distro.
```
Comment (ahopp) [inline _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:248]: Couple of notes;

- Needs a space between "[Upgrading FAQs](https://opensearch.org/faq/#c3)" and ".and if" + we either need to remove the "." or capitalize the "a". 
- Double period ("..") after backwards compatibility FAQ. 
- I'd consolidate the last two sentences. Something like "Hope this blog post helps! We plan to have a follow up post on upgrading from Elasticsearch to OpenSearch next! If you have any feedback/suggestions on new topics please reach out.

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+
+In any case you’d like to rollback your upgrade for any reason, the simplest path is to spin up an Open Distro cluster and restore using the snapshot you took earlier. You could follow the same process listed in the snapshot upgrade path above. 
+
+### Recommendation
+
+Now that you read through most of the good stuff, we would recommend these in terms of ease of upgrade:
+
+1. Snapshot Upgrade: This is the safest approach if you can spin up multiple clusters and move your customers to the new endpoint after the upgrade. we would recommend this for beginners. 
+2. Restart Upgrade: This is a moderate approach, as this would take down the whole cluster and you will only know after the whole cluster upgrade to verify everything is happy. we would recommend this for intermediate users or for people who know what they are doing.
```
Comment (ahopp) [inline _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:237]: Capital "W" in "We would recommend...".

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    kill $(cat pid)
+    ```
+
+5. Upgrade all the nodes (install OpenSearch):
+    1. Extract the zip in a new directory.
+    2. Copy over the data directory from Elasticsearch to OpenSearch or set `path.data` in `config/opensearch.yml` pointing to that directory.
+    3. Set `path.logs` in `config/opensearch.yml` pointing to a directory where you want to store logs.
+
+6. Verify existing cluster is still green and healthy.
+7. Re-enable shard allocation: Start each upgraded node: If the cluster has dedicated master nodes please start them first and make sure the master is elected before data nodes are started. You can monitor the health of the cluster via
```
Comment (ahopp) [inline _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:128]: ":" followed by another ':". A bit awkward, we may want to drop the second for a ".".

**Review by ahopp — COMMENTED** [review]
Thank Sarat! highlighted a few things, but otherwise looks good!

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown (hunk 4 above)
Comment (saratvemulapalli) [reply _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:128]: Thats a great catch, quip to markdown messed it up. It was supposed to be 2 different points.

File: _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown (hunk 2 above)
Comment (saratvemulapalli) [reply _posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown:248]: I like that. Sure just modified a little bit.

**Review by stockholmux — APPROVED** [review]
LGTM! Ship it (to staging)

**Pull request merged**